File: scripts/smoke_openai.py
# SPDX-License-Identifier: Apache-2.0
"""Separately invoked synthetic adapter smoke; never run by imports, examples or CI."""
import json
import os
import re
import signal
import sys
import time
from threading import Event
from psp_cdl_llmproxy import create_openai_chat_provider, OPENAI_CHAT_MODEL, ProviderError


def main():
    args = sys.argv[1:]
    budgets = [a for a in args if re.fullmatch(r"--budget-tokens=[1-9][0-9]*", a)]
    if len(args) != 2 or "--allow-live" not in args or len(budgets) != 1:
        print(json.dumps({"status": "not_run", "reason": "Require --allow-live and --budget-tokens=<ceiling>; supply PSP_OPENAI_API_KEY separately."}))
        return 2
    cancelled = Event()
    previous = signal.signal(signal.SIGINT, lambda *_: cancelled.set())
    try:
        provider = create_openai_chat_provider({"mode": "live", "allowLive": True, "apiKey": os.environ.get("PSP_OPENAI_API_KEY", ""), "now": lambda: int(time.time()),
                                               "complete": True, "sources": [{"id": "synthetic-smoke", "capabilities": ["is-ai-system", "cloud-processing"]}],
                                               "limits": {"maxRequestBytes": 4096, "maxResponseBytes": 8192, "maxOutputTokens": 32, "maxCalls": 1,
                                                          "budgetTokens": int(budgets[0].split("=")[1]), "timeoutMs": 15000}})
        result = provider["invoke"]({"messages": [{"role": "system", "content": "This is a synthetic adapter smoke test."}, {"role": "user", "content": "Return exactly SYNTHETIC_OK."}], "tools": []},
                                    {"deadline": int(time.time())+15, "cancelled": cancelled.is_set})
        if result.get("type") != "final" or result["text"].strip() != "SYNTHETIC_OK": raise ProviderError("SMOKE_OUTPUT_MISMATCH")
        print(json.dumps({"status": "passed", "scope": "synthetic-adapter-only", "model": OPENAI_CHAT_MODEL, "streaming": False}))
        return 0
    except Exception as exc:
        print(json.dumps({"status": "failed", "code": getattr(exc, "code", "PROVIDER_FAILED")}))
        return 1
    finally:
        signal.signal(signal.SIGINT, previous)


if __name__ == "__main__": raise SystemExit(main())

File: scripts/smoke-openai.mjs
// SPDX-License-Identifier: Apache-2.0
// Separately invoked synthetic adapter smoke. Never run by examples, imports or CI.
import {createOpenAIChatProvider,OPENAI_CHAT_MODEL} from '@psp-cdl/llmproxy';
const args=process.argv.slice(2),budgetArg=args.find(a=>a.startsWith('--budget-tokens='));
if(args.length!==2||!args.includes('--allow-live')||!budgetArg||!/^--budget-tokens=[1-9][0-9]*$/.test(budgetArg)){
  console.log(JSON.stringify({status:'not_run',reason:'Require --allow-live and --budget-tokens=<ceiling>; supply PSP_OPENAI_API_KEY separately.'}));
  process.exitCode=2;
}else{
  let cancelled=false;const stop=()=>{cancelled=true;};process.on('SIGINT',stop);
  try{
    const provider=createOpenAIChatProvider({mode:'live',allowLive:true,apiKey:process.env.PSP_OPENAI_API_KEY??'',now:()=>Math.floor(Date.now()/1000),
      complete:true,sources:[{id:'synthetic-smoke',capabilities:['is-ai-system','cloud-processing']}],
      limits:{maxRequestBytes:4096,maxResponseBytes:8192,maxOutputTokens:32,maxCalls:1,budgetTokens:Number(budgetArg.split('=')[1]),timeoutMs:15000}});
    const result=await provider.invoke({messages:[{role:'system',content:'This is a synthetic adapter smoke test.'},{role:'user',content:'Return exactly SYNTHETIC_OK.'}],tools:[]},
      {deadline:Math.floor(Date.now()/1000)+15,cancelled:()=>cancelled});
    if(result.type!=='final'||result.text.trim()!=='SYNTHETIC_OK')throw {code:'SMOKE_OUTPUT_MISMATCH'};
    console.log(JSON.stringify({status:'passed',scope:'synthetic-adapter-only',model:OPENAI_CHAT_MODEL,streaming:false}));
  }catch(e){console.log(JSON.stringify({status:'failed',code:e.code??'PROVIDER_FAILED'}));process.exitCode=1;}
  finally{process.removeListener('SIGINT',stop);}
}
